HW1/P2/P2.ipynb

```diff
@@ -0,0 +1,166 @@
+{
```

Comment:
It's good on ipython notebook, but seems difficult to add my comments on corresponding lines.
Good codes and I see you try different ways doing repartition. Thank you!
You may import a .py file instead of copy & paste all the lines into the notebook.